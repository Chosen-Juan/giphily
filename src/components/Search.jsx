import React, { Component } from 'react';
import Giphy from '../common/Giphy';
import { Gif } from './Gif';

export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      query: '',
      results: []
    };
    this.onInputChange = this.onInputChange.bind(this);
    this.renderGifs = this.renderGifs.bind(this);
  }

  onInputChange(event) {
    const query = event.target.value;
    const params = { q: query };
    Giphy.get('/search', { params })
      .then(res => this.setState({ results: res.data.data }))
      .catch(this.setState({ results: [] }));
    this.setState({ query });
  }

  renderGifs() {
    return this.state.results.map(i => {
      return <Gif gif={i} key={i.id} />;
    });
  }

  render() {
    return (
      <div>
        <input
          className='input'
          type='text'
          placeholder='Search Giphy...'
          value={this.state.query}
          onChange={this.onInputChange}
        />
        <div className='gif-container'>
          {this.renderGifs()}
        </div>
      </div>
    );
  }
}

export default Search;
