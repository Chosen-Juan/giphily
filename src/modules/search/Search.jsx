import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchGifs } from './SearchActions';
import './styles.scss';

export class Search extends Component {

  static propTypes = {
    fetchGifs: PropTypes.func.isRequired
  };

  state = {
    query: ''
  };

  onInputChange = event => {
    const query = event.target.value;
    this.setState({ query });
    this.props.fetchGifs(query);
  }

  render() {
    return (
      <div className='search-bar'>
        <input
          className='input'
          type='text'
          placeholder='Search Giphy...'
          value={this.state.query}
          onChange={this.onInputChange}
        />
      </div>
    );
  }
}

const mapStateToProps = () => {
  return {};
};

const mapDispatchToProps = {
  fetchGifs
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
