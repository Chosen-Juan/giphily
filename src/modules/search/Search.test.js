import React from 'react';
import { shallow } from 'enzyme';

import { Search } from './Search';

const fetchGifsSpy = jest.fn();
const defaultProps = {
  fetchGifs: fetchGifsSpy
};

describe('Search', () => {
  let search;

  beforeEach(() => {
    search = shallow(<Search {...defaultProps} />);
  });

  it('should match snapshot', () => {
    expect(search).toMatchSnapshot();
  });

  it('should call onChange', () => {
    const mockedEvent = { target: { value: 'mock' } }
    const div = search.find('.input');
    div.simulate('change', mockedEvent);
    expect(fetchGifsSpy).toHaveBeenCalledWith(mockedEvent.target.value);
  });

});
