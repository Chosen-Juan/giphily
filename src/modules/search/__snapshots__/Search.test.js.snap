// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div
  className="search-bar"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search Giphy..."
    type="text"
    value=""
  />
</div>
`;
